lib/elliptic/curves.js

```diff
@@ -18,8 +18,10 @@ function PresetCurve(options) {
   this.n = this.curve.n;
   this.hash = options.hash;
 
-  assert(this.g.validate(), 'Invalid curve');
-  assert(this.g.mul(this.n).isInfinity(), 'Invalid curve, G*N != O');
+  if (!skipValidation) {
+    assert(this.g.validate(), 'Invalid curve');
+    assert(this.g.mul(this.n).isInfinity(), 'Invalid curve, G*N != O');
```

Comment:
this validation is expensive, and is not necessary for preset curves

Reply:
@mvayngrib Are those validated in tests?